spec/controllers/articles_spec.rb

```diff
@@ -0,0 +1,227 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::ArticlesController, type: :controller do
+  describe "#create" do
+    subject { post :create }
+    
+    context "when no code provided" do
+      it_behaves_like "unauthorized_requests"
+    end
+   
+    context "when invalid code provided" do
+      before { request.headers["authorization"] = "Invalid token" }
+      it_behaves_like "unauthorized_requests"
+    end
+
+    context "when authorized" do
+      let(:access_token) { create :access_token }
+      before { request.headers["authorization"] = "Bearer #{access_token.token}" }
+
+      context "when invalid parameters provided" do
+        let(:invalid_attributes) do
+          {
+            data: {
+              attributes: {
+                title: "",
+                content: ""
+              }
+            }
+          }
+        end
+
+        subject { post :create, params: invalid_attributes }
+
+        it "should return 422 status code" do
+          subject
+          expect(response).to have_http_status(:unprocessable_entity)
+        end
+
+        xit "should return proper error JSON" do
+          # need to refactor test due to reliance on 
+          # depricated 'active_model_sericalizers gem
```

Comment:
will try to find a solution in future if time allows -- don't like pushing skipped tests ; )